src/libfuncs/cast.rs

```diff
@@ -57,200 +55,205 @@ pub fn build_downcast<'ctx, 'this>(
     entry: &'this Block<'ctx>,
     location: Location<'ctx>,
     helper: &LibfuncHelper<'ctx, 'this>,
-    metadata: &mut MetadataStorage,
+    _metadata: &mut MetadataStorage,
     info: &DowncastConcreteLibfunc,
 ) -> Result<()> {
     let range_check =
         super::increment_builtin_counter(context, entry, location, entry.argument(0)?.into())?;
+    let src_value: Value = entry.argument(1)?.into();
 
-    let src_type = registry.get_type(&info.from_ty)?;
-    let dst_type = registry.get_type(&info.to_ty)?;
-    let src_width = src_type
-        .integer_width()
-        .ok_or_else(|| Error::SierraAssert(SierraAssertError::Cast))?;
-    let dst_width = dst_type
-        .integer_width()
-        .ok_or_else(|| Error::SierraAssert(SierraAssertError::Cast))?;
+    if info.signature.param_signatures[1].ty == info.signature.branch_signatures[0].vars[1].ty {
+        let k0 = entry.const_int(context, location, 0, 1)?;
+        entry.append_operation(helper.cond_br(
+            context,
+            k0,
+            [0, 1],
+            [&[range_check, src_value], &[range_check]],
+            location,
+        ));
+        return Ok(());
```

Comment:
Out of curiosity, why is `cond_br` necessary when the condition is constant?

Reply:
The compiler expects to have a `cf.br` operation for each branch in the libfunc. If we didn't have this, cairo native would throw an error at runtime. LLVM should optimize it away anyways.

Reply:
Aah, thanks